questions.txt

```diff
@@ -1,51 +1,89 @@
 
-5.Can you describe your workflow when you create a web page?
+5. Can you describe your workflow when you create a web page?
 
--
+- I initially list all the requirements needed for the web page, and identify all the data that are needed (including the dependencies for each data).
+  Then I create mockups for all the possible areas that are needed for the web page.
+  After the mockups, I create migration files for the needed data, and create specifications for each models and the relationship requirements if there are any.
+  Then I proceed to coding the UI for all the requirements needed. I usually start one workflow and a finish it before I proceed to the next one.
+  One workflow for me means: fetch_data --> render UI --> process data --> re-render UI. Data fetching and processing usually takes time when the requirement is complex, and aside from coding the actual implementation, I always try and determine where to put things where, and how to make it faster and effecient.
+  I'm probably OCD because it annoys me when codes are not organized where and how it should be.
+  And finally, I take most of my time on the design because like I said, I'm OC haha. When the project has a designated UX designer, then I'm probably way faster because then I know how to design it, but if I'm the one who has to think of ways to design it, and code it, then that takes time.
+
+  So yeah, that's how I code. I'm faster on backend than on the front-end basically.
 
 6. If you have 5 different stylesheets, how would you best integrate them into the site?
 
--
+- I probably will find a way to pack them (using an asset packager or minifier) so that loading it will be much faster (especially on production).
+  If these stylesheets are used in all pages, then I will also integrate them on one file using mixins so that when the file is minified, then it's a lot smaller.
 
 7. Name 3 ways to decrease page load (perceived or actual load time).
 
--
+- Optimize fetching data.
+  - Find n+1 queries when fetching data, and refactor code to have few queries as possible.
+  - If the data can be loaded via cache (memory or browser cache), then use it to your advantage.
+- Optimize assets included on page
+  - If there are assets that are not included on the page, then remove them. We don't want extra load time to load these unnecessary js/css files.
+  - Use an asset packager to minify/reduce these files (but making sure it still retains the information in it)
+  - If there are images, if possible, delay loading and load only when needed/viewed.
+- Create the illusion that the page is fast
+  - This is more on perceived faster page load. If you have lots of data to render, apply pagination that will load the data in bits.
+  - Use ajax to your advantage in CRUD data. This will minimize additional page reload, and much fun to use.
 
```

Comment:
This is actually 3 ways, but it seems a lot. But to sum it up:
- Optimize fetching data
- Optimize assets included on page
- Create the illusion that the page is fast (using javascript to delay load everything else)